### PR TITLE
refactor(cash): implement better UX patterns

### LINKED ISSUE #81: (feature request): put a "notes" field on the Cash Payment form
All of our other forms allow for custom notes - Vouchers, Patient Invoices, etc.  It would be both expected and useful to have notes associated with a cash payment.  That way, any additional information (for example, the service they should be sent to) can be included on the Cash Payment form.

### LINKED ISSUE #90: (feature request) Put a "clear" button on the Cash Payments form to clear selection
The cash payments page currently allows a user to refresh the debtor via the `bhFindPatient` component.  This leads to a corner cases:  what happens when an invoice is selected and a patient is changed?

It is also of note that the refresh button is a poor user experience in this page for two reasons:
 1. It is non-standard - Patient Invoicing does not use it, why should Cash do differently?
 2. While finding by ID, if you click "Search" twice, on a fast network, the patient will appear and the user will click "refresh" accidentally.

For these reasons, I recommend that we move to a "Clear" button like on the Patient Invoice page.

### PR DESCRIPTION
**Pull request by jniles**
This PR implements:
 1. A "clear" button on the cash payments page that allows a user to clear all cash data.
 2. A notes field on the cash payments page that allows a user to input custom notes.  They will be printed out on the POS receipt.

Closes #90.  Closes #81.

### CODE REVIEW

File: client/src/partials/cash/cash.html [hunk 1]
```diff
@@ -12,30 +13,30 @@
     <div class="row">
 
       <!-- Switch Cashbox Form and Transfer -->
-      <div class="col-md-2">
+      <div class="col-md-2 col-lg-3">
```
Comment (sfount) [inline client/src/partials/cash/cash.html]: :+1:

File: client/src/partials/cash/cash.html [hunk 2]
```diff
@@ -86,15 +86,15 @@
...
+                        <th style="width: 25%;" translate>TABLE.COLUMNS.REFERENCE</th>
+                        <th style="width: 50%;" translate>TABLE.COLUMNS.DATE</th>
+                        <th style="width: 25%;" class="text-right" translate>TABLE.COLUMNS.AMOUNT</th>
                       </tr>
                     </thead>
                     <tbody>
 
                       <!-- show a list of previous invoices to pay again -->
-                      <tr ng-repeat="invoice in CashCtrl.payment.invoices track by invoice.uuid">
+                      <tr ng-repeat="invoice in CashCtrl.payment.invoices | orderBy:'date' track by invoice.uuid">
```
Comment (sfount) [inline client/src/partials/cash/cash.html]: 👍 🎈

File: client/src/partials/cash/cash.js [hunk 3]
```diff
@@ -56,7 +57,7 @@ function CashController(Cash, Cashboxes, AppCache, Currencies, Exchange, Session
     vm.timestamp = new Date();
 
     /* This is the actual payment form */
-    settupPayment();
```
Comment (sfount) [inline client/src/partials/cash/cash.js]: @sfount. This is not how you spell setup.

Comment (jniles) [inline client/src/partials/cash/cash.js]: It's German, anyway.

**Pull request merged**